modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java

```diff
@@ -232,16 +232,16 @@ public void setExtendedProperties(
 				objectRelationshipElementsParser.parse(
 					objectRelationship, entry.getValue());
 
-			if (!nestedObjectEntries.isEmpty()) {
-				DefaultObjectEntryManager defaultObjectEntryManager =
-					DefaultObjectEntryManagerProvider.provide(
-						_objectEntryManagerRegistry.getObjectEntryManager(
-							objectDefinition.getStorageType()));
-
-				defaultObjectEntryManager.disassociateRelatedModels(
-					objectDefinition, objectRelationship, getPrimaryKey(entity),
-					relatedObjectDefinition, userId);
-			}
+			DefaultObjectEntryManager defaultObjectEntryManager =
+				DefaultObjectEntryManagerProvider.provide(
+					_objectEntryManagerRegistry.getObjectEntryManager(
+						objectDefinition.getStorageType()));
+
+			defaultObjectEntryManager.disassociateRelatedModels(
+				_getDefaultDTOConverterContext(
+					objectDefinition, getPrimaryKey(entity), null),
```

Comment:
You are invoking many times to thje `getPrimaryKey` method. Please, move it to a variable